ntrans.py

```diff
@@ -1,107 +1,75 @@
-        source_target_pairs.append((source_string, str(translated_string).lower()))
+    for n in ngrams.keys():
+        for enum, source_ngram in enumerate(ngrams[n], start=1):
+            print(f"Translating {n}-gram no. {enum} / {len(ngrams[n])}")
+            target_ngram = str(
+                translator.translate(source_ngram, "Swedish")
+            ).lower()  # TODO: Target language should be variable-based
+            source_target_pairs.append((source_ngram, target_ngram))
 
     create_csv_file(source_target_pairs)
 
 
-def count_ngram_frequency(ngram_strings, desired_data_size):
-    x_most_common_ngrams = collections.Counter(ngram_strings).most_common(desired_data_size)
-    return [ngram[0] for ngram in x_most_common_ngrams]
-
-
-def create_ngrams(corpus_data, ngram_size, desired_data_size):
-    """
-    Strips string from special characters, creates tuple n_grams, deletes tuples
-    with numbers, joins tuples into strings
+def read_ngram_files(user_desired_ngrams=[2, 3, 4, 5, 6], data_size=500):
     """
+    Reads N-gram files depending on which N-grams the user wants to output.
 
-    # Removes punctuation and special characters from string.
-    text_without_punctuation = re.sub(r"[^\w']+", " ", corpus_data).lower()
-
-    # TODO: Make the .replaces more effective
-    text_without_punctuation = text_without_punctuation.replace(" '", "'")
-    text_without_punctuation = text_without_punctuation.replace(" n't", "n't")
+    Appends the N-grams to a dict with a size of data_size per N-gram (specified by user).
+    """
 
-    # TODO: Put ngram function in for loop to include all N-grams that the user wants.
-    # TODO: Switch place of ngram creation and joining ngram_tuples into strings, since joining ngram_tuples takes a long time.
-    # Splits string into N-grams and adds them to tuple_list.
-    generated_ngram_tuples = nltk.ngrams(text_without_punctuation.split(), ngram_size)
+    ngrams = {2: [], 3: [], 4: [], 5: [], 6: []}
 
-    # Joins tuples into strings, and deletes strings with numbers.
-    # TODO: Takes long time. Possible to make faster?
-    ngram_strings = []
-    print('Joining n-gram tuples to strings')
-    for generated_ngram_tuple in generated_ngram_tuples:
-        joined_tuple = " ".join(generated_ngram_tuple)
-        if not any(char.isdigit() for char in joined_tuple):
-            ngram_strings.append(joined_tuple)
+    for n in user_desired_ngrams:
+        with open(f"./ngrams/{n}-grams.csv") as ngram_file:
+            read_csv = csv.reader(ngram_file)
+            for enum, row in enumerate(read_csv):
+                if enum == data_size:
+                    break
+                ngrams[n].append(row[0])
 
-    count_ngram_frequency(ngram_strings, desired_data_size)
-    # # return machine_translate_ngrams(count_ngram_frequency(ngram_strings, desired_data_size))
+    machine_translate_ngrams(ngrams)
 
 
 """
 CODE TESTING:
 """
```

Comment:
This is not a valid docstring either, because it's not at the start of the file:

```suggestion
# CODE TESTING:
```